File: backend/src/main/java/com/tasklion/backend/features/tasklionAccount/service/TasklionAccountService.java
package com.tasklion.backend.features.tasklionAccount.service;

import com.tasklion.backend.common.constant.TasklionUserRole;
import com.tasklion.backend.common.model.KeyValueModel;
import com.tasklion.backend.features.tasklionAccount.TasklionAccount;
import com.tasklion.backend.features.tasklionAccount.model.ChangePasswordRequestModel;
import com.tasklion.backend.features.tasklionAccount.model.TasklionAccountDetailsModel;
import com.tasklion.backend.features.tasklionAccount.model.TasklionAccountModel;

public interface TasklionAccountService {

    TasklionAccount saveTasklionAccount(TasklionAccount tasklionAccount, TasklionUserRole tasklionUserRole);

    TasklionAccountModel getTasklionAccount(String username);

    TasklionAccountDetailsModel getTasklionAccountDetail(String username);

    boolean isExists(KeyValueModel<String> requestModel);

    void updateTasklionAccountStatus(String username, String status);

    TasklionAccountDetailsModel updateTasklionAccountDetail(String username, TasklionAccountDetailsModel tasklionAccountDetailsModel);

    boolean changePassword(String username, ChangePasswordRequestModel changePasswordRequestModel);
}

File: frontend/src/app/modules/tasklion-account/services/tasklion-account.service.ts
import {Injectable} from '@angular/core';
import {Observable} from "rxjs";
import {ApiEndpointConstant} from 'src/app/shared/constants/api-endpoint.constant';
import {ApiResponseModel} from 'src/app/shared/models/api/api-response.model';
import {KeyValueModel} from 'src/app/shared/models/key-value.model';
import {ApiService} from 'src/app/shared/services/api/api.service';
import {TasklionAccountDetailModel} from "../models/tasklion-account-detail.model";

@Injectable({
  providedIn: 'root'
})
export class TasklionAccountService extends ApiService {

  public isExists(fieldValueModel: KeyValueModel): Observable<ApiResponseModel<any>> {
    return this.post<ApiResponseModel<any>>(ApiEndpointConstant.IS_TASKLION_ACCOUNT_EXISTS, fieldValueModel);
  }

  public getTasklionAccount(username: string): Observable<ApiResponseModel<any>> {
    return this.get<ApiResponseModel<any>>(`${ApiEndpointConstant.TASKLION_ACCOUNT}/${username}`);
  }

  public updateTasklionAccount(username: string, tasklionAccountDetail: TasklionAccountDetailModel): Observable<ApiResponseModel<any>> {
    return this.put<ApiResponseModel<any>>(`${ApiEndpointConstant.TASKLION_ACCOUNT}/${username}`, tasklionAccountDetail);
  }

  public changePassword(username: string, changePasswordModel: KeyValueModel): Observable<ApiResponseModel<any>> {
    return this.post<ApiResponseModel<any>>(`${ApiEndpointConstant.CHANGE_PASSWORD}/${username}`, changePasswordModel);
  }

  public updateTasklionAccountStatus(username: string, status: string): Observable<ApiResponseModel<any>> {
    return this.put<ApiResponseModel<any>>(`${ApiEndpointConstant.TASKLION_ACCOUNT}/${username}${ApiEndpointConstant.STATUS}`, status);
  }

}
